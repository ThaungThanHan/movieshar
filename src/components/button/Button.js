const Button = ({label}) => {
    return(
        <div data-testid="button" className="buttonClass" style={{opacity:1,border:"1px solid red",width:30,height:30}}>
            This is {label}
        </div>
    )
}

export default Button;