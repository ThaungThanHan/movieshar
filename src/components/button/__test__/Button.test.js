import React from 'react';
import ReactDOM from 'react-dom';
import Button from "./../button";
import {isTSAnyKeyword} from "@babel/types";
import {render} from "@testing-library/react";
import '@testing-library/jest-dom'
import renderer from "react-test-renderer";
// npm install @testing-library/react @testing-library/jest-dom react-test-renderer --save-dev
// Using ReactDOM
it("renders without crashing",()=>{             // check if renders or not. Normall, component is not rendered until called. So, to test if it renders, we can create
                                                // DOM element div and binds the <Button/> component to it.
    const div = document.createElement("div");
    ReactDOM.render(<Button/>,div);
    ReactDOM.unmountComponentAtNode(div);
})

//Using testing-library/react and jest-dom
it("renders correctly",()=>{
    const {getByTestId} = render(<Button label="Click me please!" />)
    expect(getByTestId('button')).toHaveTextContent("Click me please!")
})

// it("renders correctly",()=>{    // check jest-dom npm package for more methods of checking!
//     const {getByTestId} = render(<Button label="SAVE" />)
//     expect(getByTestId('button')).toBeVisible();
// })
// toHaveClass
// toHaveStyle
// toBeVisible
// 

it("matches snapshot",()=>{         // if snapshot doesn't match, we gotta decide whether to update the snapshot or change the code! right change? Update snapshot!
    const tree = renderer.create(<Button label="hello"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})