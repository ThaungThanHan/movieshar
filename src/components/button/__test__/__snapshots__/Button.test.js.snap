// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="buttonClass"
  data-testid="button"
  style={
    Object {
      "border": "1px solid red",
      "height": 30,
      "opacity": 1,
      "width": 30,
    }
  }
>
  This is 
  hello
</div>
`;
